docs/DeveloperGuide.md

```diff
@@ -221,7 +222,65 @@ for subsequent prompts.
+
+Inside each of these 'create command' methods, the following generalised procedure to create an object of the subclass 
+of `Command` is carried out:
+1. (For commands that expect an action keyword (e.g. `/list`, `/new`)) The action keyword is parsed and determined.
+    - If the action keyword is invalid, an `InvalidCommandException` is thrown and the parsing is aborted.
+2. Depending on the action keyword (or lack thereof), the number of arguments are checked.
+    - If insufficient or too many arguments are provided in the user input, an `InvalidCommandException` is thrown
+   and the parsing is aborted.
+3. A new object of the subclass of `Command` is created and if the object is successfully constructed with no errors,
+it is returned to `Parser#parseUserInput()`.
+
+**(Step 11)** The object created is then returned to `WerkIt#startContinuousUserInput()`.
+
+<span class="box info"> 🧾 (About the sequence diagram) Stricly speaking, the object is returned right after whichever 
+'create command' method is invoked. However, to improve the readability of the diagram, only one return line is shown,
+since all alternate paths will return an object that is a subclass of the `Command` class.</span>
+
+**(Step 12)** The final step of this section is to invoke the `Command#execute()` method, which will in turn call the
+overridden `execute()` method of the subclass of `Command`.
+- For example, if the user input is `workout /new push up /reps 10`, the created `WorkoutCommand` object is upcasted
+to `Command` when returned to `WerkIt#startContinuousUserInput()`, but when `newCommand.execute()` is called,
+`WorkoutCommand#execute()` is called.
+
+Thereafter, the appropriate procedures are taken to complete the task requested by the user. The various procedures
+are explained in later sections of this developer guide.
+
+#### Illegal Characterse & Phrases
+Some symbols and phrases are reserved for use by the application and thus are not allowed to be used by the user
+in his/her inputs to avoid any potential instabilities when processing his/her inputs.
+
+| Illegal Character/Phrase | Purpose in Application                                                                                                                                                                |
+|--------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| The pipe character &#124; | Used as a delimiter in the app data files to separate the various data. Allowing the user to use delimiters in their plan names may cause issues when storing them in the data files. |
+| The phrase 'rest day' | Used as an indicator that a particular day in the user's schedule does not have a plan in it. Allowing the user to name a plan as 'rest day' may cause issues when displaying the schedule. |
+
```

Comment:
This is a very good table and explanation. A further suggestion is perhaps can say "However, even if those characters are entered, the app has taken measures to handle these inputs ..."

Reply:
Thanks for your feedback. I have included an additional statement in Commit f9ab5cd.